2018/proposals/gensim Aneesh Joshi Similarity Learning.md

```diff
@@ -0,0 +1,399 @@
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> MV-LSTM </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.5818 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6452 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.5988 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> DRMM </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6107 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6621 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6195 </td>
+  </tr>
+  <tr>
+     <td align="center", bgcolor=#eeeeee> aNMM </td>
+     <td align="center", bgcolor=#eeeeee> 0.6160 </td>
+     <td align="center", bgcolor=#eeeeee> 0.6696 </td>
+     <td align="center", bgcolor=#eeeeee> 0.6297 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> DUET </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6065 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6722 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6301 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> MatchPyramid </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6317 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6913 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6434 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> DRMM_TKS </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6458 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6956 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6586 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> K-NRM </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6268 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6693 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6256 </td>
+  </tr>
+</table>
+
+Along with the loss:
+
+![alt-text](https://github.com/faneshion/MatchZoo/blob/master/docs/_static/images/matchzoo.wikiqa.loss.png)
+
+Free datasets for Similarity Learning will also be added to [gensim-data](https://github.com/RaRe-Technologies/gensim-data) :
+
+* [Quora Duplicate Questions](https://data.quora.com/First-Quora-Dataset-Release-Question-Pairs) (already in gensim-data)
+* [WikiQA](https://www.microsoft.com/en-us/download/details.aspx?id=52419)
+* [Ubuntu Dialogue Corpus](https://github.com/rkadlec/ubuntu-ranking-dataset-creator)
+
+# Deliverables
+
+The whole project can be broken down into 4 milestones
+1. Benchmark and Evaluation of existing models
+2. Get one model working end-to-end
+3. Repeat the same for remaining models
+4. Disseminate models
+
+### Community Bonding Period (April 23 to May 14)
+* Design the api for the models in gensim (Basically what goes where and how the user calls it)
+* Discuss and update the list of models to be implemented.
+* Start verifying and benchmarking all the mentioned models using [MatchZoo](https://github.com/faneshion/MatchZoo)
+* Publish results and comparisons as a blog
+
+**Milestone 1 complete!**
+
+### Week 1 (May 14 to May 20)
+* Find and understand existing data preprocessing code in gensim

```

Comment:
We almost have no preprocessing code, your input is 2 documents (as list of tokens, that already processed)

Reply:
@menshikh-iv 
The preprocessing code I was referring to was:
1. `scan_vocab`
2. `prepare_vocab`

Basically, dropping words, tokenizing, giving ids to words.

Do you think I should still change it?